### ISSUE TITLE
Automatically Create Passwords and Use the Website, Favorite Name Generator.

### ISSUE THREAD

**Issue by CerebralFreeze** [keepassxreboot/keepassxc#4739]
## Summary: 
**Easier, Prettier, and Tougher Login/Password Combos Automatically Created.**
Users frequently rely on KeePassXC when they are creating a new login/password for a new online account.  The following steps will make the entire process easier for users.  KeePassXC provides a great, fake friendly username and an unbreakable password.  After the online account is made with these new details, KeePassXC then automatically creates a new entry with the new Title, new Login, Password and URL filled out.  Finally, the password in the clipboard is replaced with the previous contents.

## Details
The first step is to create the login and password.  I usually want an unbreakable password that is about 20 characters long with symbols.  On the other hand, I like to create a Username that is more friendly.  It's still fake but it's friendly because I might be using it for a forum.  I use this popular website, https://www.fakenamegenerator.com/ to create a real-sounding username.  (It's a pretty awesome website that provides realistic identity details.)  I've written an AutoHotkey script so that a password generator creates an 15-20 character-long password with upper/lower case letters, numbers, and low symbols.  This password gets saved in the clipboard.  Then, I have my script open up the website, fakenamegenerator.com.

I type the new _long_ full name as the username and I press Control+V to paste the complex password as the new password.  I do the exact thing with KeePassXC.  I have the original clipboard contents replace the password 1 minute later.

## Context
This new feature isn't urgent but I think it will be a nice little bonus that will make the program more user friendly and facilitate the entire new-account-making process that people hate so much.

## Please Help Me Answer This Question.  I Made a Large Bet with a Friend.
I have to jog 5 miles if I lose.  I know this sounds like a stupid question but please indulge me.  I know this program is open-source so anyone can use it but does a particular person "own" the rights to the script?  Or, is it owned by the public?  Can anyone just take the source code, alter it quite a bit, and start selling it for a major profit?

Or, is the the script owned by a person or a group of people who can later change their mind and privatize and close the source code?

**Comment by droidmonkey** [comment]
First, KeePassXC is licensed as GPL-v3 which means anyone can copy the source, make changes, and redistribute. You must make fully available the modified code and cannot change the license. You're welcome to sell GPL licensed software, but because you have to make the source available it doesn't stop anyone from distributing it for free.

Second, username generators have been proposed in the past and we have decided as a project not to support that feature. It's a niche feature and adds unnecessary complexity.

For what it's worth, our browser extension handles entry creation automatically.

**Comment by CerebralFreeze** [comment]
Thanks, droidmonkey for the response!

Been a little busy to talk to ya'.  I had just lost that bet about who was right about how open-source works so I had to jog those 5 miles!  Sometimes, you bring bad news.  ;)  On the bright side, I got some exercise but 5 miles is A LOT for a fattie like me.  You almost gave me a heart attack.  

I also know you are hard at work too so I don't want to bother you that much.  I really don't understand how you do all the work you do except you must love what you're doing.  So, it's not really "work"?  Is that the secret?

With respect to the original technical question about the definition of open-source, I guess you guys can't just sell something, by revealing the main source code but hiding additional code that you created?  The entire code, even the modified, I guess has to be revealed?  

I was thinking that this stinks for you, guys.  I mean this genuinely but this is one of the first open-source programs that really is pretty awesome and it can turn into a really kick-ass password manager.  Are you saying that you, guys, can't just sell it to Microsoft without revealing the entire source code?!?  

If so, the ultimate possible end goal for this project is for everyone in the world to end up using the KeePassXC manager which would be awesome but no monetary compensation for the people who put so much time, blood, and tears into it.  That doesn't sound right to me for some reason.

I mean this genuinely too but you, guys, deserve something.
